### PR TITLE
fix reproject raster layer using gdal

### PR DESCRIPTION
**Pull request by Gustry**

### CODE REVIEW

File: safe/gisv4/raster/reproject.py [hunk 1]
```diff
@@ -36,18 +45,91 @@ def reproject(layer, output_crs, callback=None):
...
     """
+    # I have a problem reprojecting with PyQGIS. I'm using GDAL.
+    # ET 26/09/16
+
+    # Fixme : Manage callback in these functions.
+    return _reproject_gdal(layer, output_crs, callback)
+
+
+def _reproject_gdal(layer, output_crs, callback=None):
+    """Reproject using GDAL command line.
```
Comment (ismailsunni) [inline safe/gisv4/raster/reproject.py]: Add param and type

File: safe/gisv4/raster/reproject.py [hunk 2]
```diff
@@ -36,18 +45,91 @@ def reproject(layer, output_crs, callback=None):
...
+    try:
+        reprojected.keywords = layer.keywords
+    except AttributeError:
+        raise KeywordNotFoundError
+
+    return reprojected
+
+
+def _reproject_pyqgis(layer, output_crs, callback=None):
+    """Not working properly for now. Need more tests.
```
Comment (ismailsunni) [inline safe/gisv4/raster/reproject.py]: Add param and type.

File: safe/gisv4/vector/reproject.py [hunk 3]
```diff
@@ -33,8 +35,8 @@ def reproject(layer, output_crs, callback=None):
     :type output_crs: QgsCoordinateReferenceSystem
```
Comment (ismailsunni) [inline safe/gisv4/vector/reproject.py]: Complete param and type.

**Comment by Gustry** [comment]
@ismailsunni to process a raster, we have 3 ways : 
- use the QGIS API only
- use the GDAL API
- use the GDAL command line

I tried to use to use the QGIS api, because it's better if we can use QgsRasterLayer everywhere. If I use GDAL API or GDAL command line, I need to work with a file on the file-system.
But it's not working properly. That's why I added this kind of wrapper to switch easily when I will find why it's not working with PyQGIS. The raster processing in pyqgis is not very well documented instead of GDAL.
So in the meantime, I added the GDAL command line to do the job.

File: safe/gisv4/vector/reproject.py (hunk 3 above)
Comment (Gustry) [inline safe/gisv4/vector/reproject.py]: Just expand the code, you will see it's there.

Comment (ismailsunni) [inline safe/gisv4/vector/reproject.py]: Ah, my bad

**Comment by Gustry** [comment]
We are not going to reproject raster. I'm going to close this PR and make a new one with only the code we need.

**Pull request closed without merging**